xbmc/addons/AddonDatabase.cpp

```diff
@@ -1011,7 +1023,7 @@ bool CAddonDatabase::GetBlacklisted(std::set<std::string>& addons)
   return false;
 }
 
-bool CAddonDatabase::BlacklistAddon(const std::string& addonID)
+bool CAddonDatabase::SetUpdateRuleForAddon(const std::string& addonID, AddonUpdateRule updateRule)
```

Comment:
`AddUpdateRuleForAddon`

Reply:
there is an index set on the new table `addonID, updateRule` so no duplicate entries are possible.
but we can have multiple entries per addon. think there is no update functionality needed here